src/gennylib/include/gennylib/parallel.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_5129031F_B241_46DD_8285_64596CB0C155_INCLUDED
+#define HEADER_5129031F_B241_46DD_8285_64596CB0C155_INCLUDED
+
+#include <boost/throw_exception.hpp>
+#include <boost/exception/exception.hpp>
+#include <boost/core/noncopyable.hpp>
+#include <boost/log/trivial.hpp>
+
+#include <algorithm>
+#include <iterator>
+#include <variant>
+#include <thread>
+#include <mutex>
+#include <vector>
+#include <deque>
+
+namespace genny {
+
+/**
+ * Class that represents a bundle of exceptions thrown together after
+ * ending parallel execution.
+ */
+class ParallelException : public std::exception {
+public:
+
+    ParallelException(std::deque<std::exception_ptr>&& exceptions) {
+        _caughtExceptions = std::move(exceptions);
```

Comment:
init statement?

```c++
ParallelEx(deque&& exc)
: _caught{std::move(exc)} { .... }
```

(That `std::move` isn't strictly necessary of course but it's become an accepted piece of ceremony)